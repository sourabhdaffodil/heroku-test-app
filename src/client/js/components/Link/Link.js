import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const A = styled.a`
  color: inherit;
  text-decoration: none;

  &.link {
    color: ${props => props.theme.color()}
  }
`;

const Link = ({ url, className, children }) => (
  <A href={url} className={className} target="_blank" rel="noreferrer noopener">
    { children }
  </A>
);

Link.propTypes = {
  url: PropTypes.string.isRequired,
  className: PropTypes.string,
  children: PropTypes.any.isRequired,
};

Link.defaultProps = {
  className: '',
};

export default Link;
