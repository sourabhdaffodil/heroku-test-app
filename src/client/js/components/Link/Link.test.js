import React from 'react';
import { shallow } from 'enzyme';
import Link from './Link';

describe('Link component', () => {
  it('Must render properly', () => {
    const container = shallow(
      <Link url="sample">
        Sample
      </Link>,
    );

    expect(container).toMatchSnapshot();
  });
});
