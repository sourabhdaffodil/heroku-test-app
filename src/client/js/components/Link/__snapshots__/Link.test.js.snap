// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component Must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Link
    className=""
    url="sample"
  >
    Sample
  </Link>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": "Sample",
      "className": "",
      "href": "sample",
      "rel": "noreferrer noopener",
      "target": "_blank",
    },
    "ref": null,
    "rendered": "Sample",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": "Sample",
        "className": "",
        "href": "sample",
        "rel": "noreferrer noopener",
        "target": "_blank",
      },
      "ref": null,
      "rendered": "Sample",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
